### ISSUE TITLE
Standardizing the formats of libp2p specs

### ISSUE THREAD

**Issue by Mikerah** [libp2p/specs#160]
I noticed that you folks at Protocol Labs have been focusing on cleaning up and improving the libp2p specs, in addition to adding new specs. Great work! However, one major thing I noticed across all the specs in this repo is the lack of structure across all the specs. Every spec writer has their own way of writing specs and as such we end up with a lot of different styles of specifications. This will bring a lot of problems as more community members start implementing their own libp2p compliant modules in various programming languages.

I suggest that libp2p specs have a standard way that spec writers write specs. I would suggest taking a look at how the IETF and W3C write specifications that are standardized.

**Comment by raulk** [comment]
Thanks for bringing this up. I agree. This wad evidenced recently with confusion about how to phrase certain items for effectiveness and precision, the aspects they should cover, and the viewpoints they should be expressed from (implementers, users, etc.) I’m making resolving this a priority for myself, in the pursuit of more mature processes for libp2p engineering.

**Comment by zinid** [comment]
I agree with @Mikerah: the specs are unreadable. Compare for example [RELOAD p2p protocol](https://tools.ietf.org/html/rfc6940) with the stuff in this repo: night and day.
